ci/mkpipeline.py

```diff
@@ -104,10 +109,62 @@ def main() -> int:
             print(
                 "Repository glue code has changed, so the trimmed pipeline below does not apply"
             )
-            trim_tests_pipeline(copy.deepcopy(pipeline), args.coverage)
+            trim_tests_pipeline(copy.deepcopy(pipeline), args.coverage, args.sanitizer)
         else:
             print("--- Trimming unchanged steps from pipeline")
-            trim_tests_pipeline(pipeline, args.coverage)
+            trim_tests_pipeline(pipeline, args.coverage, args.sanitizer)
+
+    if args.sanitizer != "none":
+        pipeline.setdefault("env", {})["CI_SANITIZER"] = args.sanitizer
+
+        def visit(step: dict[str, Any]) -> None:
+            # ASan runs are slower ...
+            if "timeout_in_minutes" in step:
+                step["timeout_in_minutes"] *= 3
```

Comment:
Should we do this with coverage builds as well? Not with factor 3 but probably some `1.x`

Reply:
We already do that for coverage with factor 3.